### PR TITLE
Upgrade nigoapi 0.1.0

### PR DESCRIPTION
**Pull request by juldrixx**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | yes
| Related tickets | -
| License         | Apache 2.0

### What's in this PR?

Upgrade of NiGoApi from v0.0.10 to v0.1.0.

### Why?

To use the latest API from NiFi 2.0.0-M2.

### Checklist

- [X] Implementation tested
- [X] Error handling code meets the [guideline](docs/error-handling-guide.md)
- [X] Logging code meets the guideline
- [X] User guide and development docs updated (if needed)
- [x] Append changelog with changes

### CODE REVIEW

**Comment by juldrixx** [comment]
@mh013370 Should I remove `MaximumEventDrivenThreadCount` from v1 CRD or just do it like that?

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +16,8 @@
 
 ### Deprecated
 
+- [PR #422](https://github.com/konpyutaika/nifikop/pull/422) - **[Operator/NifiCluster]** `MaximumEventDrivenThreadCount` is no longer configurable in NiFi.
```
Comment (mh013370) [inline CHANGELOG.md:19]: This should be removed from the nifi-cluster helm chart as well.

File: site/docs/5_references/1_nifi_cluster/2_read_only_config.md [hunk 2]
```diff
@@ -144,7 +144,7 @@ readOnlyConfig:
 |Field|Type|Description|Required|Default|
 |-----|----|-----------|--------|--------|
 |maximumTimerDrivenThreadCount|int32|define the maximum number of threads for timer driven processors available to the system.|No|10|
-|maximumEventDrivenThreadCount|int32|define the maximum number of threads for event driven processors available to the system.|No|1|
+|maximumEventDrivenThreadCount|int32|define the maximum number of threads for event driven processors available to the system (@DEPRECATED).|No|1|
```
Comment (mh013370) [inline site/docs/5_references/1_nifi_cluster/2_read_only_config.md:147]:
```suggestion
|maximumEventDrivenThreadCount|int32|define the maximum number of threads for event driven processors available to the system (@DEPRECATED. This has no effect starting with NiFiKOp v1.9.0 or later).|No|1|
```

**Comment by mh013370** [comment]
> @mh013370 Should I remove `MaximumEventDrivenThreadCount` from v1 CRD or just do it like that?

Removing it is a breaking change, but we could document that from nifikop v1.9.0+ it has no effect and will be removed in later CRD versions.

**Comment by mh013370** [comment]
Is this update compatible with NiFi 1.x NiFi deployments? Or is this PR staged for 2.0?

File: site/docs/5_references/1_nifi_cluster/2_read_only_config.md (hunk 2 above)
Comment (juldrixx) [reply site/docs/5_references/1_nifi_cluster/2_read_only_config.md]: Should be okay now.

File: CHANGELOG.md (hunk 1 above)
Comment (juldrixx) [reply CHANGELOG.md:19]: Should be okay now.

**Comment by juldrixx** [comment]
> Is this update compatible with NiFi 1.x NiFi deployments? Or is this PR staged for 2.0?

Older version than 1.18 won't fully work. I tested in 1.16 and 1.17 and the operator can't create the registry client so it can't deploy dataflow from it but it can do the rest. And in >=1.18, it works fine. I can create user, group, parameter context, registry client, dataflow. I updated the documention to reflect it.

This was to be expected, as version 1.23 of the api ended up working with version 2.0.0-M2.

**Review by mh013370 — APPROVED** [review]
LGTM thanks for updating the compatibility matrix!

**Pull request merged**